cpp/src/quantiles/quantile.cu

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <memory>
+#include <thrust/functional.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <cudf/copying.hpp>
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/table/row_operators.cuh>
+#include <cudf/table/table_device_view.cuh>
+#include <cudf/types.hpp>
+#include <cudf/utilities/traits.hpp>
+#include <quantiles/quantiles_util.hpp>
+#include "cudf/utilities/error.hpp"
+
+namespace cudf {
+namespace experimental {
+namespace detail {
+namespace {
+
+using ScalarResult = double;
+
+struct quantile_functor
+{
+    template<typename T, typename... Args>
+    std::enable_if_t<not std::is_arithmetic<T>::value, std::unique_ptr<scalar>>
+    operator()(Args&&... args)
+    {
+        CUDF_FAIL("Only numeric types are supported in quantiles.");
+    }
+
+    template<typename T>
+    std::enable_if_t<std::is_arithmetic<T>::value, std::unique_ptr<scalar>>
+    operator()(column_view const& input,
+               double percent,
```

Comment:
But it's not a percent, right? A percent is between 0 and 100.
```suggestion
               double q,
```

Reply:
Indeed. This was mentioned in the previous pr, but I missed this function when renaming.